website_event_snippet_calendar/controllers/main.py

```diff
@@ -39,13 +42,24 @@ def days_with_events(self, start, end):
                 now += one_day
         return [Date.to_string(day) for day in days]
 
+    def get_event_type_from_parameters(self, parameters=None):
+        if not parameters:
+            return None
+        event_type = parameters.get("type", None)
+        if event_type:
+            try:
+                event_type = int(event_type) or None  # 0 is not a valid ID
+            except ValueError:
+                event_type = None
```

Comment:
@Camille0907 I don't understand what you try to achieve here:
```
>>> int(0)
0
>>> int('0')
0
```
can you pls clarify?

Reply:
Maybe you can give some examples of `event_type`?

Reply:
`event_type` is supposed to be the id of a `event.type` (<=> an "Event Template") existing in the system. Events can be linked to an event template, so we want here to have the possibility to filter on the events linked to a specific "Event template" : in this case the ID of the event template has to be stored in this variable `event_type`.
This variable comes from the URL of the page, from a GET parameter : `https://my-site/specific-path?type=5`. Thus we can't be sure to have a valid ID : it can be something else than a number (caught by the `except`), or it can be 0 (IDs begin from 1) : since 0 is not valid we want the same behavior as in the `except`.
So : 
- if event_type is not a number -> we go in the `except` -> None
- if event_type is 0 (or '0') -> int(event_type) = 0 <=> False -> we go in the second part of the `or` -> None
- if event_type is a number >= 1 -> int(event_type) != 0 <=> True -> we stay in the first part of the `or` -> event_type stores the valid id

I hope I'm clear and answering to your remark